zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/ElasticsearchEngineClientIT.java

```diff
@@ -0,0 +1,163 @@
+    doReturn(indexName).when(descriptor).getIndexName();
+    doReturn("newProperties.json").when(descriptor).getMappingsClasspathFilename();
+
+    // when
+
+    elsEngineClient.putMapping(
+        descriptor,
+        IOUtils.resourceToString(
+            "newProperties.json", StandardCharsets.UTF_8, getClass().getClassLoader()));
+
+    // then
+    final var index = elsClient.indices().get(req -> req.index(indexName)).get(indexName);
+
+    assertThat(index.mappings().properties().get("email").isKeyword()).isTrue();
+  }
+
+  @Test
+  void shouldCreateComponentTemplateCorrectly() throws IOException {
+    // given
+    final var componentTemplateName = "test";
+    final var descriptor = mock(ComponentTemplateDescriptor.class);
+    when(descriptor.getTemplateName()).thenReturn(componentTemplateName);
+    when(descriptor.getTemplateClasspathFileName()).thenReturn("componentTemplate.json");
+
+    // when
+    elsEngineClient.createComponentTemplate(descriptor);
+
+    // then
+    final var templates =
+        elsClient.cluster().getComponentTemplate(req -> req.name(componentTemplateName));
+
+    assertThat(templates.componentTemplates().size()).isEqualTo(1);
+
+    final var componentTemplate = templates.componentTemplates().getFirst();
+    assertThat(componentTemplate.name()).isEqualTo(componentTemplateName);
+
+    final var json = getClass().getClassLoader().getResourceAsStream("componentTemplate.json");
+    final var expectedComponentTemplate =
+        elsEngineClient.deserializeJson(ComponentTemplateNode._DESERIALIZER, json);
+    assertThat(componentTemplate.componentTemplate().toString())
+        .isEqualTo(expectedComponentTemplate.toString());
+  }
+
+  @Test
+  void shouldCreateIndexTemplateCorrectly() throws IOException {
+    // given
+    final var descriptor = mock(IndexTemplateDescriptor.class);
+    doReturn("index_name").when(descriptor).getIndexName();
+    doReturn("test*").when(descriptor).getIndexPattern();
+    doReturn("alias").when(descriptor).getAlias();
+    doReturn(Collections.emptyList()).when(descriptor).getComposedOf();
+
+    final var templateName = "template_name";
+    doReturn(templateName).when(descriptor).getTemplateName();
+    doReturn("mappings.json").when(descriptor).getMappingsClasspathFilename();
+
+    // when
+    elsEngineClient.createIndexTemplate(descriptor);
+
+    // then
+    final var indexTemplates =
+        elsClient.indices().getIndexTemplate(req -> req.name(templateName)).indexTemplates();
+
+    assertThat(indexTemplates.size()).isEqualTo(1);
+
+    final var retrievedTemplate = indexTemplates.getFirst().indexTemplate().template();
+    final var template =
+        elsEngineClient.deserializeJson(
+            IndexTemplateSummary._DESERIALIZER,
+            getClass().getClassLoader().getResourceAsStream("mappings.json"));
+    assertThat(retrievedTemplate.mappings().toString()).isEqualTo(template.mappings().toString());
```

Comment:
can we also assert it inherits the component template properties (settings) it's composed of?